libr/util/str.c

```diff
@@ -1505,9 +1506,48 @@ R_API char *r_str_escape_utf32be(const char *buf, int buf_size, bool show_asciid
 	return r_str_escape_utf (buf, buf_size, R_STRING_ENC_UTF32BE, show_asciidot, esc_bslash, false);
 }
 
-// JSON has special escaping requirements
-// TODO: merge with r_str_escape_utf() and r_str_byte_escape() using RStrEsc
-R_API char *r_str_escape_utf8_for_json(const char *buf, int buf_size) {
+R_API char *r_str_encoded_json(const char *buf, int buf_size, int encoding) {
+	if (!buf) {
+		return NULL;
+	}
+	char *encodedString;
+
+	if (encoding == PJ_ENCODING_BASE64) {
+		encodedString = r_base64_encode_dyn (buf, buf_size);
+	} else if (encoding == PJ_ENCODING_HEX || encoding == PJ_ENCODING_ARRAY) {
+		int loop = 0;
+		int i = 0;
+		size_t len = buf_size < 0 ? strlen (buf) : buf_size;
+		size_t increment = encoding == PJ_ENCODING_ARRAY ? 4 : 2;
+		size_t new_sz = (len * increment) + 1;
+		const ut8 *format = encoding == PJ_ENCODING_ARRAY ? "%03u," : "%02X";
+
+		encodedString = malloc (new_sz);
+		if (!encodedString) {
+			printf ("can't malloc\n");
+			return NULL;
+		}
+		while (buf[loop] != '\0' && i < (new_sz - 1)) {
+			snprintf (encodedString + i, sizeof (encodedString), format, (ut8) buf[loop]);
+			loop += 1;
+			i += increment;
+		}
+		if (encoding == PJ_ENCODING_ARRAY) {
+			// get rid of the trailing comma
+			encodedString[i - 1] = '\0';
```

Comment:
use `es` instead of camelcased var name. al si i think i can be 0 in this line and harm the heap